### PR TITLE
rizin: init at unstable-2021-01-13

### PR DESCRIPTION
**Pull request by Pamplemousse**
###### Motivation for this change

[rizin](https://rizin.re/) is a fork of [radare2](https://radare.org/n/).
And the software powering [Cutter](https://cutter.re/).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Signed-off-by: Pamplemousse <xav.maso@gmail.com>

### CODE REVIEW

**Comment by Pamplemousse** [comment]
This is a "best effort" to get something out there and gather feedback.

`rizin` build is meant to be easier than `radare`'s, but I am not sure how well all the features are supported by the current expression.

**Comment by Pamplemousse** [comment]
Poke @makefu , @7c6f434c , @Mic92 .

File: pkgs/development/tools/analysis/rizin/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+  src = fetchFromGitHub {
+    owner = "rizinorg";
+    repo = "rizin";
+    rev = "ca2471ae9596176c656f065ed74c480f13784e1b";
+    sha256 = "0qigy1px0jy74c5ig73dc2fqjcy6vcy76i25dx9r3as6zfpkkaxj";
+  };
+
+  postPatch = let
+    capstone = fetchFromGitHub {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/rizin/default.nix:19]: Please move this to a let in before mkDerivation.

File: pkgs/development/tools/analysis/rizin/default.nix [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  propagatedBuildInputs = [ libzip xxHash ];
+
+  meta = {
+    description = "unix-like reverse engineering framework and commandline tools";
+    homepage = "https://rizin.re/";
+    license = stdenv.lib.licenses.lgpl3Only;
+    maintainers = with stdenv.lib.maintainers; [ pamplemousse ];
+    platforms = with stdenv.lib.platforms; linux;
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/rizin/default.nix:51]:
```suggestion
  meta = with lib; {
    description = "unix-like reverse engineering framework and commandline tools";
    homepage = "https://rizin.re/";
    license = licenses.lgpl3Only;
    maintainers = with maintainers; [ pamplemousse ];
    platforms = with platforms; linux;
  };
```

File: pkgs/development/tools/analysis/rizin/default.nix [hunk 3]
```diff
@@ -0,0 +1,52 @@
+{ stdenv
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/rizin/default.nix:1]:
```suggestion
{ stdenv
, lib
```

File: pkgs/development/tools/analysis/rizin/default.nix [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  postPatch = let
+    capstone = fetchFromGitHub {
+      owner = "aquynh";
+      repo = "capstone";
+      # version from $sourceRoot/shlr/Makefile
+      rev = "4.0.2";
+      sha256 = "0y5g74yjyliciawpn16zhdwya7bd3d7b1cccpcccc2wg8vni1k2w";
+    };
+  in ''
+    mkdir -p build/shlr
```
Comment (SuperSandro2000) [inline pkgs/development/tools/analysis/rizin/default.nix:27]:
```suggestion
    mkdir shlr
```

File: pkgs/development/tools/analysis/rizin/default.nix [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+stdenv.mkDerivation {
+  pname = "rizin";
+  version = "unstable-2021-01-13";
+
+  src = fetchFromGitHub {
+    owner = "rizinorg";
+    repo = "rizin";
+    rev = "ca2471ae9596176c656f065ed74c480f13784e1b";
+    sha256 = "0qigy1px0jy74c5ig73dc2fqjcy6vcy76i25dx9r3as6zfpkkaxj";
+  };
```
Comment (Mic92) [inline pkgs/development/tools/analysis/rizin/default.nix:16]: They still rely on git to set versions etc:

https://github.com/rizinorg/rizin/blob/d4964454765539c61cce7f3f845f8d7fe3a3f076/Makefile#L60

In our other radare2 build we generate those automatically.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which does not build all packages (e.g. lumo, tensorflow or pytorch)
If you find some bugs or got suggestions for further things to search or run please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 109305` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
1 package built:
</summary>
<ul>
<li>rizin</li>
</ul>
</details>

**Comment by Mic92** [comment]
also included in https://github.com/NixOS/nixpkgs/pull/118041

**Pull request closed without merging**